metricbeat/scripts/mage/docs_collector.go

```diff
@@ -120,6 +123,34 @@ func testIfDocsInDir(moduleDir string) bool {
 	return true
 }
 
+// compile and run the seprate go script to generate a list of default metricsets.
+// This is done so a compile-time issue in metricbeat doesn't break the docs build
+func getDefaultMetricsets() (map[string][]string, error) {
+
+	runpaths := []string{
+		mage.OSSBeatDir("scripts/msetlists/main.go"),
+		mage.XPackBeatDir("scripts/msetlists/main.go"),
+	}
+
+	var masterMap = make(map[string][]string)
+	for _, dir := range runpaths {
+		rawMap, err := sh.OutCmd("go", "run", dir)()
+		if err != nil {
+			return nil, err
+		}
+		var msetMap = make(map[string][]string)
+		err = json.Unmarshal([]byte(rawMap), &msetMap)
+		if err != nil {
+			return nil, err
+		}
+		for k, v := range msetMap {
+			masterMap[k] = v
+		}
+	}
+
+	return masterMap, nil
+}
```

Comment:
Ok, let's go with this approach, it'd be similar to other x-pack scripts being called from OSS build scripts.